ssl/statem/statem_clnt.c

```diff
@@ -3354,8 +3384,12 @@ WORK_STATE tls_prepare_client_certificate(SSL *s, WORK_STATE wst)
             }
             s->rwstate = SSL_NOTHING;
         }
-        if (ssl3_check_client_certificate(s))
+        if (ssl3_check_client_certificate(s)) {
+            if (s->post_handshake_auth == SSL_PHA_REQUESTED) {
```

Comment:
I will check.

Reply:
Not so sure. If `ssl3_check_client_certificate()` returns 1 (i.e. there's a good certificate), then it doesn't go into `WORK_MORE_B`, which this PR does not change. I think the real problem is that if `ssl3_check_client_certificate()` returns 0, it will fall into `WORK_MORE_B` state which won't do the transition of `post_handshake_auth` to `SSL_PHA_REQUESTED`

Reply:
If `WORK_FINISHED_STOP` is not returned during PHA, then the client will expect there to be  Certificate message following the CertificateRequest, which won't be happening with PHA.

Reply:
Right....so the the `WORK_MORE_B` state code needs to change to return `WORK_FINISHED_STOP` in the event of `ssl3_check_client_certificate()` returning 0 and we're in PHA?

Reply:
Yes, otherwise, the client waits for a Certificate message that never comes, and doesn't respond. I have additional commits with fixes forthcoming.